deepinv/datasets/datagenerator.py

```diff
@@ -152,9 +153,66 @@ def close(self):
             self.hd5 = None
 
 
+def collate(dataset: Dataset):
```

Comment:
Please can you add a docstring and return type to explain what this does? e.g. why it returns none, or a func...